discord/guild.py

```diff
@@ -3397,8 +3405,19 @@ async def ban(
             You do not have the proper permissions to ban.
         HTTPException
             Banning failed.
+        TypeError
+            You specified both ``delete_message_days`` and ``delete_message_seconds``.
         """
-        await self._state.http.ban(user.id, self.id, delete_message_days, reason=reason)
+        if delete_message_days is not MISSING and delete_message_seconds is not MISSING:
+            raise TypeError('Cannot mix delete_message_days and delete_message_seconds keyword arguments.')
+
+        if delete_message_days is not MISSING:
+            delete_message_seconds = delete_message_days * 86400  # one day
```

Comment:
Would we want to raise a warning here? is there some way for us to do that currently, or would a new utility need to be made.

Reply:
The mechanism for raising a warning here was pretty standard, I did something similar with `request_offline_members` in `Client.__init__`.

You can add `warnings.warn(msg, DeprecationWarning, stacklevel=2)`

Reply:
Done.